source: disable check for URL marking at configure time

abderrahim opened the pull request:
This check is triggered by plugins that have URLs (or URL fragments) as part of their ref, such as cargo. These plugins currently resort to only translating part of the URL and appending the rest later. Dropping the check allows these plugins to call mark_download_url() or translate_url() a second time when tracking.
    
It also allows plugins to postpone calling them until fetch(), which is the behaviour the check was supposed to prevent. This is still considered wrong, but will no longer trigger an AssertionError.

This MR also removes the `_configure()` and `_get_configuring()` methods of `Plugin` as they are no longer needed.

abderrahim committed:
source: drop check for URL marking at configure time

This check is triggered by plugins that have URLs (or URL fragments)
as part of their ref, such as cargo. These plugins currently resort to
only translating part of the URL and appending the rest
later. Dropping the check allows these plugins to call
mark_download_url() or translate_url() a second time when tracking.

It also allows plugins to postpone calling them until fetch(), which
is the behaviour the check was supposed to prevent. This is still
considered wrong, but will no longer trigger an AssertionError.

nanonyme commented:
Seems to me the entire marked URL's mechanism should be dropped if this is done. `self.__marked_urls` doesn't seem to be used anywhere else.

abderrahim committed:
plugin: drop _configure() and _get_configuring()

They are no longer needed now that we don't enforce URL marking at configure
time.

abderrahim commented:
superseded by #1899
